src/main/app/index.js

```diff
@@ -266,10 +319,37 @@ class App {
       ])
     }
 
-    if (_openFilesCache.length) {
-      this._openFilesToOpen()
+    // If followSystemTheme=true, wait for nativeTheme to settle before creating windows.
+    // On some systems (especially Linux), shouldUseDarkColors may not immediately
+    // reflect the OS theme when themeSource is first set to 'system'.
+    // Waiting for the 'updated' event ensures windows are created with the
+    // correct background color, preventing a white flash on startup.
+    if (followSystemTheme) {
+      let windowCreated = false
+
+      const createWindow = () => {
+        if (windowCreated) return
+        windowCreated = true
+
+        if (_openFilesCache.length) {
+          this._openFilesToOpen()
+        } else {
+          this._createEditorWindow()
+        }
+      }
+
+      // Wait for 'updated' event (fires if theme changes)
+      nativeTheme.once('updated', createWindow)
+
+      // Fallback timeout in case 'updated' never fires (no theme change)
+      setTimeout(createWindow, 150)
     } else {
-      this._createEditorWindow()
+      // Not following system theme - create windows immediately
+      if (_openFilesCache.length) {
+        this._openFilesToOpen()
+      } else {
+        this._createEditorWindow()
```

Comment:
[Nitpick] Consider declaring the function creatWindow above this if/else block so it can be used in both if/else branches instead of re-declaring code